art/estimators/object_detection/utils.py

```diff
@@ -88,3 +92,50 @@ def convert_pt_to_tf(y: List[Dict[str, np.ndarray]], height: int, width: int) ->
         y[i]["labels"] = y[i]["labels"] - 1
 
     return y
+
+
+def cast_inputs_to_pt(
+    x: Union[np.ndarray, "torch.Tensor"],
+    y: Optional[List[Dict[str, Union[np.ndarray, "torch.Tensor"]]]] = None,
+) -> Tuple["torch.Tensor", List[Dict[str, "torch.Tensor"]]]:
+    """
+    Cast object detection inputs `(x, y)` to PyTorch tensors.
+
+    :param x: Samples of shape NCHW or NHWC.
+    :param y: Target values of format `List[Dict[str, Union[np.ndarray, torch.Tensor]]]`, one for each input image.
+                The fields of the Dict are as follows:
+
+                - boxes [N, 4]: the boxes in [x1, y1, x2, y2] format, with 0 <= x1 < x2 <= W and 0 <= y1 < y2 <= H.
+                - labels [N]: the labels for each image.
+    :return: Object detection inputs `(x, y)` as tensors.
+    """
+    import torch
+
+    # Convert images into tensor
+    if isinstance(x, np.ndarray):
+        x_tensor = torch.from_numpy(x)
+    else:
+        x_tensor = x
+
+    # Convert labels into tensor
+    if y is not None and isinstance(y, list) and isinstance(y[0]["boxes"], np.ndarray):
+        y_tensor = []
```

Comment:
Would we be able to remove the "type: ignore" in line 139 if we add more specific typing here in line 122 like
```suggestion
        y_tensor: List[Dict[str, "torch.Tensor"]] = []
```
If yes, should we move this initialization before this if-block?

Reply:
The `mypy` issue is actually due to the `None` type that `y` can be. The `type: ignore` was actually here before, I've just abstracted this segment of code out (since it is now reused) and made it a function. 

Attempting to make the typing changes to correct this needs to go a couple levels upwards which can makes thing overcomplicated.

Reply:
This should now be resolved